Project.toml

```diff
@@ -1,11 +1,12 @@
 name = "Agents"
 uuid = "46ada45e-f475-11e8-01d0-f70cc89e6671"
 authors = ["George Datseris", "Tim DuBois", "Aayush Sabharwal", "Ali Vahdati", "Adriano Meligrana"]
-version = "5.15.2"
+version = "5.16.0"
 
 [deps]
-CommonSolve = "38540f10-b2f7-11e9-35d8-d573e4eb0ff2"
+Arrow = "69666777-d1a9-59fb-9406-91d4454c9d45"
```

Comment:
Now that we have moved in Julia 1.9 we should be using the extensions framework to make Arrow an extension instead of a formal dependency. We have already done this with Makie so it should be easy to apply here. The key is to make a function `write_arrow` without any arguments in the file that is inside agents, and then define a method for this function with arguments inside the `ext` folder.

If how extensions work isn't clear I'll do it later.

Reply:
I've had a stab at this. Tests pass locally but I still expect the CI tests on Windows to fail so somebody needs to look at it on their local Windows machine to find out what the problem is.

I haven't implemented any package extensions so far, so please @Datseris have a look at what I did and if I missed something.

Moved the `AgentsVisualizations` and `AgentsOSMVisualizations` extensions into their own subfolders under `ext/` to make sure that the extensions files don't get mixed up. (See [here](https://pkgdocs.julialang.org/dev/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions)) that this is possible.)

Reply:
Oh, and I removed `arrow_integration.jl` in the AgentsIO submodule for now. It would only make sense to have this as a package extension as well then. (Same applies to the JLD2 integration but I won't touch that in this PR.)

Reply:
the problem on Windows is related to permissions: it fails when calling `rm(adata.arrow)` since the os refuses to remove the file, didn't still find any solution and I'm not really understanding why that doesn't happen with the .csv file created and removed some lines before this. Is it possible that the arrow file is still open when trying to remove it?

Reply:
Interestingly it doesn't fail with `rm(adata.arrow)`

Reply:
> Is it possible that the arrow file is still open when trying to remove it?

Hm, shouldn't be the case but what do I know about Windows stuff... Worth a try at least?